controller/controller.go

```diff
@@ -226,6 +229,39 @@ func ListenAndServeEx(host string, port int, dir string, ln *net.Listener, http
 	return server.ListenAndServe(host, port, protected, handler, opened, closed, ln, http)
 }
 
+func (c *Controller) watchGC() {
+	var elapsed uint64
+	t := time.NewTicker(time.Second)
+	defer t.Stop()
+
+	for range t.C {
+		elapsed++
+		c.mu.RLock()
+		autoGC := c.config.AutoGC
+		c.mu.RUnlock()
+
+		func() {
+			if autoGC == 0 {
+				return
+			}
+
+			if c.stopWatchingAutoGC {
+				return
+			}
+
+			if elapsed < autoGC {
+				return
+			}
+
+			c.mu.RLock()
```

Comment:
We don't need to execute the runtime.GC() and FreeOSMemory() calls inside of a lock.